Display Certifications

boredazfcuk opened the pull request:
Overlays the certifications over the posters/banners/fanart for easy recognition of age certification. Works with GB/US for movies and US for TV shows.

## Description
This is just a UI tweak that allows the age certifications to be displayed on the poster/banners/fanart, reminiscent of VHS days. It adds two new options in the Skin Settings. One for displaying US/GB Movie age ratings and another for displaying US age ratings for TV Shows.

## Motivation and context
This change make adult oriented films plainly obvious, and should prevent people from picking inappropriate movies to watch with their children.

## How has this been tested?
I've been running this since March 2019, and originally forked it here: https://github.com/boredazfcuk/skin.estuary. I did not realise at the time that the original repository was not maintained, and never got around to submitting it to XBMC. I've now upgraded to Matrix so have merged the changes to the latest Estuary Skin.

## What is the effect on users?
Users can easily see age ratings. e.g. U/PG/12A/15/18  for UK and G/PG/R/NC-17 for USA. Also TV-Y/TV-Y7/TV-14/TV-MA, for TV shows.

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ X] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

boredazfcuk committed:
Display Certifications

Overlays the certifications over the posters/banners/fanart for easy recognition of age certification. Works with GB/US for movies and US for TV shows.

boredazfcuk committed:
Display Certifications

Overlays the certifications over the posters/banners/fanart for easy recognition of age certification. Works with GB/US for movies and US for TV shows.

CastagnaIT commented:
just a first thought,
here you assume that images are always shown by skin, so if a skin do not provide the support to show the image or a listitem do not have the images?
could be appropriate to manage age ratings in some way as media-flag instead/or also above the images?
maybe also sound good to centralise management in a c++ class to manage _all_ use cases of each country? if you manage all in skin xml, seems to do fragmented maintenance, if something change also external suppliers will also need to adapt each skins

gade01 commented:
Thanks for the PR.
I didn't look at the skin side of things.

New strings should only be added to en_gb file.
Weblate will add strings to all other language files.

boredazfcuk committed:
Removed translations that were not required

boredazfcuk committed:
Removed translations that were not required

boredazfcuk committed:
Update strings.po

boredazfcuk committed:
Update strings.po

boredazfcuk committed:
Merge branch 'master' into master

boredazfcuk commented:
> @boredazfcuk this needs a rebase

Done.

boredazfcuk commented:
> just a first thought, here you assume that images are always shown by skin, so if a skin do not provide the support to show the image or a listitem do not have the images? could be appropriate to manage age ratings in some way as media-flag instead/or also above the images? maybe also sound good to centralise management in a c++ class to manage _all_ use cases of each country? if you manage all in skin xml, seems to do fragmented maintenance, if something change also external suppliers will also need to adapt each skins

Hi, this is way more than than what I implemented. It wasn't a mechanism for all skins to implement the feature. It was just an Esturary skin modification I made.

I submitted the original PR to https://github.com/phil65/skin.estuary, as that was the original repo for the skin at the time. I presume it's abandoned, as I submitted it in March 2019 and it's still not had a reply from the repo maintainer.

I didn't add anything other than GB/US for movies and US for TV shows as, if memory serves me correctly, the metadata was not available for other countries. I'm not sure if that's still true, as it was nearly two years ago when I made this.

CastagnaIT commented:
I am not here to say you are wrong, but i'm puzzled for me kodi currently provides an unsuitable rating system way, and i think it should be changed/improved

ok for estuary only, but i was thinking broader,
currently kodi have `ListItem.mpaa` and currently this is the only way to set the mature rating for a video content

but this property is often used by each video addons of each country video content, then the value set often is not related to MPAA standard, it is clear that here kodi has a big lack...

IMO a first thing to do should be integrate/refactor the kodi mature rating system feature to accept all rating systems available (rename it only in more generic way etc...) and not limit Kodi to set and show MPAA only,
each country has its own scale of rating system to measure maturity, here an example of a list by country:
https://www.primevideo.com/help/ref=atv_nb_lcl_en_US?ie=UTF8&nodeId=GFGQU3WYEG6FSJFJ
https://en.wikipedia.org/wiki/Motion_picture_content_rating_system

with the view that this should be improved in the future to be accept all type of mature rating system,
as you are doing we would have to add a lot of images in the skin for the rating system of each country, perhaps is better find an alternative way to avoid make a lot of images? maybe a simple Label control?

and you have add the `gb` but use BBFC rating system, why it is add for MPAA?
as we only have MPAA for now we should only add the MPAA standard?

another thing is not clear to me because you have make a distinction between "tvshow" rating system and "movie" rating system, both use the same rating standard, is there a reason why this further distinction should be made?

boredazfcuk commented:
> I am not here to say you are wrong, but i'm puzzled for me kodi currently provides an unsuitable rating system way, and i think it should be changed/improved
> 
> ok for estuary only, but i was thinking broader, currently kodi have `ListItem.mpaa` and currently this is the only way to set the mature rating for a video content
> 
> but this property is often used by each video addons of each country video content, then the value set often is not related to MPAA standard, it is clear that here kodi has a big lack...

The "ListItem.MPAA" thing is just what the field is called in Kodi. It's an unfortunate, legacy name from when the certification system was implemented. Ideally they should have called it ListItem.Certification, but it's not, so here we are. It's not for the sole use of MPAA certifications, it for the use of any country's rating system. 

> IMO a first thing to do should be integrate/refactor the kodi mature rating system feature to accept all rating systems available (rename it only in more generic way etc...) and not limit Kodi to set and show MPAA only, each country has its own scale of rating system to measure maturity, here an example of a list by country: https://www.primevideo.com/help/ref=atv_nb_lcl_en_US?ie=UTF8&nodeId=GFGQU3WYEG6FSJFJ https://en.wikipedia.org/wiki/Motion_picture_content_rating_system
> 
> with the view that this should be improved in the future to be accept all type of mature rating system, as you are doing we would have to add a lot of images in the skin for the rating system of each country, perhaps is better find an alternative way to avoid make a lot of images? maybe a simple Label control?

There are skins which implement certifications as labels. however, I prefer the images as they feel more authentic.

> and you have add the `gb` but use BBFC rating system, why it is add for MPAA? as we only have MPAA for now we should only add the MPAA standard?

I'm not sure I understand... gb is Great Britain and the certification system used there is the British Board of Film Classification (BBFC). I also added MPAA certifications as an alternative option, as it's the default certification region set by the Universal Movie Scraper and also the TMDb scraper. Enabling the option in Kodi should have an immediate visible effect.

> another thing is not clear to me because you have make a distinction between "tvshow" rating system and "movie" rating system, both use the same rating standard, is there a reason why this further distinction should be made?

TV and movies don't use the same classifications. In the US, movies are certified as G, PG, PG-13, R, NC-17 and NR which I've added for Not Rated (a.k.a Unrated). TV certifications in the US are rated TV-Y, TV-Y7, TV-G, TV-PG, TV-14 and TV-MA.

In GB, the certifications are U, Uc, PG, 12, 12A, 15, 18 & R18.  However, there isn't an official certification system for TV shows.

> you use the `certification` word in different place sound a bit weird, is not more appropriate use word like "content rating/mature rating/rating system"?

I use the word "certification" for clarity. Both the Universal Movie Scraper and the TMDb Scraper refer to it as the "certification". "Ratings" is a separate Kodi feature for information related to how "good" a movie is. If you go to `Settings` > `Skin Settings` > `Choose rating to display for media items` and set it to "Rating" you will see a label overlay on your posters with the movie's score... e.g. 8.3 for The Shawshawn Redemption... This is why I purposefully try avoid using the word "Rating" when referring to "Certifications". It could cause confusion between the two types of information.

boredazfcuk committed:
Unused GB TV Certifications Removed

boredazfcuk committed:
Merge branch 'master' of https://github.com/boredazfcuk/xbmc

CastagnaIT commented:
> The "ListItem.MPAA" thing is just what the field is called in Kodi. It's an unfortunate, legacy name from when the certification system was implemented.

Then should be revisited to prevent misunderstandings, because currently in the wiki/doxygen it is clearly specified that the use is for MPAA only, and here you are add more than MPAA

> I use the word "certification" for clarity. Both the Universal Movie Scraper and the TMDb Scraper refer to it as the "certification".

this is not about Scrapers, the world "certification" is not so common, the major VOD websites or services use word like "mature rating" "rating system" or also "restrictions" or "age recommendations". Also in my country "certification" is never used and sound wrong also when translated

> I'm not sure I understand... gb is Great Britain and the certification system used there is the British Board of Film Classification (BBFC). I also added MPAA certifications as an alternative option, as it's the default certification region set by the Universal Movie Scraper and also the TMDb scraper. Enabling the option in Kodi should have an immediate visible effect.

here I was referring to the fact to the current `ListItem.mpaa` use case that it is specified for ".mpaa" (as property name said) and you have add also BBFC that is not MPAA. But `ListItem.mpaa` can be set by all addons without using scrapers
(here too same reason, `ListItem.mpaa` and all related things should be revisited)

> TV and movies don't use the same classifications.

as far as i can see it is not really necessary to make the distinction, all codes are never the same
this will unnecessarily complicate things for other countries to be add,
i really you want to dinstinct them you have to find a solution for countries that do not use distinctions between "tvshow" and "movie"

> "Ratings" is a separate Kodi feature for information related to how "good" a movie is.

i know it, but "Rating" and "Mature Rating" are different things, i doubt they are confused with each other

> There are skins which implement certifications as labels. however, I prefer the images as they feel more authentic.

preferring is not always the best thing to follow,
you are not considering the amount of images needed for all countries (where other countries can not have images) and how to categorise them (folders),
here are categorised in two separate folders "tvshow" and "movie", this means that with all other countries we will have to double the images unnecessarily?
if we think to add images for all countries we have to add more than 110 images! then if we have to force this distinction will be needed double the images will result like more than 200 images!
to me this raises the alarm bell and asks myself is it appropriate to do this?

From my part this PR need some changes
I ask other colleagues a feedback on my concerns

KarellenX commented:
FWIW, there is a current conversation regarding changes to MPAA ratings in Kodi and scrapers...
https://forum.kodi.tv/showthread.php?tid=365698

boredazfcuk commented:
> Then should be revisited to prevent misunderstandings, because currently in the wiki/doxygen it is clearly specified that the use is for MPAA only, and here you are add more than MPAA

Where do you see that? Because the Wiki clearly states that MPAA is for country specific certifications: https://kodi.wiki/view/NFO_files/Movies
```
NFO (xml) Tag | Required | Multiple | Notes
<mpaa></mpaa> | No | No | Country specific mpaa rating system. Check with skin author which prefix is required to diplay your country certification flags
```
Ideally this would be relabelled to <Certification>, but that seems like a widely impactful change for no benefit (beyond cosmetic)... OK, you could make this change in Kodi + the Universal Movie Scraper + the TMDb scraper, but this field is used for localised certifications by countless other apps... Ember Media Manager, TinyMediaManager, MediaElch, Jellyfn, Emby.

> this is not about Scrapers, the word "certification" is not so common, the major VOD websites or services use word like "mature rating" "rating system" or also "restrictions" or "age recommendations". Also in my country "certification" is never used and sound wrong also when translated.

Not so common? I did not choose to call it "Certification". I followed the precedence which Kodi and many others have already set out. It's called "Certification" in the Universal Movie Scraper, The Movie Database Scraper and TMDb scrapers, all of which are developed by TeamKodi. The same is true for metadata managers such as MediaElch and tinyMediaManager. It's referred to as "Certification" on imdb.com and tmdb.com.

> here I was referring to the fact to the current `ListItem.mpaa` use case that it is specified for ".mpaa" (as property name said) and you have add also BBFC that is not MPAA. But `ListItem.mpaa` can be set by all addons without using scrapers (here too same reason, `ListItem.mpaa` and all related things should be revisited)

To be frank, attempting to change this is like pissing into the wind. Yeah, OK, Kodi could introduce a new "Certification" field (or Parental/Maturity Rating or whatever else pleases you). But I highly doubt that imdb.com/tmdb.com/MediaElch/tinyMediaManager/etc are all going to follow suit. Why would they? They already have a field which serves the purpose. It's just unfortunately named.

> as far as i can see it is not really necessary to make the distinction, all codes are never the same this will unnecessarily complicate things for other countries to be add, i really you want to dinstinct them you have to find a solution for countries that do not use distinctions between "tvshow" and "movie"

I've made the distinction because it makes sense from a future proofing perspective. Some countries may have a movie certification of `PG` and a TV certification of `PG`, but with differing images. The images need to be named exactly the same certification so that they are loaded by the skin, so in this case we would have a file conflict. (There can't be two different images with the same name in a single folder) so the path needs to be differnet.

It's better to make this distinction now, than to run into this problem after 30 countries have been added. 

> you are not considering the amount of images needed for all countries (where other countries can not have images) and how to categorise them (folders), here are categorised in two separate folders "tvshow" and "movie", this means that with all other countries we will have to double the images unnecessarily? if we think to add images for all countries we have to add more than 110 images! then if we have to force this distinction will be needed double the images will result like more than 200 images! to me this raises the alarm bell and asks myself is it appropriate to do this?

I did actually consider that. It's why I made the PNG images from scratch. To make sure they were as small as possible. The images for US TV+Movies and UK movies is 134k combined. That averages to 67k per country, even with 100 countries added, the images would be less than 10Mb. That's pretty much nothing in modern terms.

jjd-uk commented:
First of all, thank you for the PR & contribution.

I've not looked at the detail of the implementation or done any testing yet, so my only comment for now is that the images should not be bundled into Estuary.

What I believe should happen with the images is that they should be handled similar manner to Studio logo's, where an image resource addon `resource.images.studios.white` is used.  There is an image resource addon already available in repo, see https://github.com/xbmc/repo-resources/tree/leia/resource.images.classificationicons.colour and dev's repo is at https://github.com/wyrm65/resource.images.classificationicons.colour

However before we ask you to consider changing this, I'd like a comment from the lead maintainer of Estuary who is @ronie on the principle of adding this before we get into the details. If the principle is deemed ok then the discussion can happen of whether we should wait for the outcome of https://forum.kodi.tv/showthread.php?tid=365698

Personally I'd be ok with this proceeding if changed to use a image resource addon. Since things proceed slowly in the world of Kodi, so I'd always prefer us to use something that could be got working now, rather than waiting for some hypothetical change in the future.

boredazfcuk commented:
> FWIW, there is a current conversation regarding changes to MPAA ratings in Kodi and scrapers... https://forum.kodi.tv/showthread.php?tid=365698
> 
> @pkscout @wyrm65

Great read. Brought back a lot of memories from when I made the skin modification and the issues I had with my DB. I forgot that I needed to run hundreds of SQL queries to get the Certifications to a standard... what a ball ache that was.

jjd-uk commented:
One other thing. It would be helpful if you added screenshots to show the placement of the images in the various views.

boredazfcuk commented:
> What I believe should happen with the images is that they should be handled similar manner to Studio logo's, where an image resource addon `resource.images.studios.white` is used. There is an image resource addon already available in repo, see https://github.com/xbmc/repo-resources/tree/leia/resource.images.classificationicons.colour and dev's repo is at https://github.com/wyrm65/resource.images.classificationicons.colour

The UK certification images shown in that repo are the BBFC images (e.g. https://github.com/wyrm65/resource.images.classificationicons.colour/blob/master/resources/UK%2015.png) you can tell by the way they have "BBFC" in the background. They are subject to copyright as per the BBFC website: https://www.bbfc.co.uk/about-us/news/media-centre

>"BBFC age rating symbols

>BBFC age ratings are under copyright and should only be used against film, video or online content classified by the BBFC. However, if you are a journalist and would like to use the symbols in an article, or on a page layout, then we will happily grant you permission to do this – please just get in touch with the Communications Team. If you have any questions about how the symbols should be used, then please do get in touch."

I'm guessing the owner of the repo doesn't have distribution rights to those images? This is why I created my own (the GB ones have "KODI" in the background instead of BBFC) instead of using the ones from the BBFC, as I didn't want to be in breach of copyright, or put Kodi in breach of copyright for distributing copyrighted material if this PR did make it into the full release. I also created the US ones myself it took me hours to find the specific font they used and to verify it was free to distribute.

boredazfcuk commented:
> One other thing. It would be helpful if you added screenshots to show the placement of the images in the various views.

Movies currently Looks like this:

![1](https://user-images.githubusercontent.com/14807258/146183465-69865033-cb4a-4eae-bf28-9ebd029920ed.png)
![2](https://user-images.githubusercontent.com/14807258/146183481-8a77cb96-ec56-44de-aca5-619b170ada58.png)
![3](https://user-images.githubusercontent.com/14807258/146183490-e421dcac-d246-4f21-9221-7aade17d04bf.png)
![4](https://user-images.githubusercontent.com/14807258/146183500-c9272265-c12e-47c9-a4d3-32d3c7f02936.png)
![5](https://user-images.githubusercontent.com/14807258/146183505-a19589b7-59f9-439d-9a9b-fafa332f7506.png)
![6](https://user-images.githubusercontent.com/14807258/146183514-fd25434a-8e69-4129-98bd-c776f5e72c1a.png)
![7](https://user-images.githubusercontent.com/14807258/146183526-9e2c9baf-fff8-49ab-8d78-d9b4bca1c7c7.png)
![8](https://user-images.githubusercontent.com/14807258/146183538-cc703bc3-f69f-48c7-9a3e-567b186b538c.png)

jjd-uk commented:
Images with copyright are ok to use I think as long as they are free to use, that Kodi is compatible with any license, and any necessary credit is given. That page also states this:

> You can download a range of high quality images, which also include our age rating symbols and shapes – **so please do feel free to use them wherever you might need**

So I think their only concern is to not use their images to misreport the rating of content, so if content is rated 18 by them, then you can't use the U rating image for example.. But I'm no expert on these things so you maybe right that it's best to play safe.

CastagnaIT commented:
> Not so common? I did not choose to call it "Certification". I followed the precedence which Kodi and many others have already set out. It's called "Certification" in the Universal Movie Scraper, The Movie Database Scraper and TMDb scrapers, all of which are developed by TeamKodi. The same is true for metadata managers such as MediaElch and tinyMediaManager. It's referred to as "Certification" on imdb.com and tmdb.com.

i am not referring to "scrapers" addons, I base myself on what is used today in VOD services providers,
if the other maintainers are ok for use "certification" I'll live with it, but will not be accepted by me, as i do not find it appropriate.

> To be frank, attempting to change this is like pissing into the wind. Yeah, OK, Kodi could introduce a new <Certification> field 

yes it is clear this will be deprecated in favour of a new more appropriate parameter

> Where do you see that? Because the Wiki clearly states that MPAA is for country specific certifications: https://kodi.wiki/view/NFO_files/Movies

for addon cases:
https://kodi.wiki/view/InfoLabels
https://codedocs.xyz/AlwinEsch/kodi/group__python___info_tag_video.html#gafdddb4046567a6dc1ea91f8ab4d424ee
currently this confuses everyone because the property is named as "mpaa" and the descriptions OFC is referred to MPAA only

On the NFO link the description say:
`Country specific mpaa rating system. Check with skin author which prefix is required to diplay your country certification flags`
for me the explanation is ambiguous, because is written `mpaa` `rating system` then not a generic rating system
of course all this will have to be changed before or after

boredazfcuk commented:
> i am not referring to "scrapers" addons, I base myself on what is used today in VOD services providers, if the other maintainers are ok for use "certification" I'll live with it, but will not be accepted by me, as i do not find it appropriate.

...and I'm not particularly interested what the VOD providers call it. They don't have authority in the matter... I know that the scrapers, metadata websites, nfo applications, etc aren't the authority either, however, the long-established, general consensus (of the English speaking world at least) is that these logos are considered certifications, so I accept that...

> yes it is clear this will be deprecated in favour of a new more appropriate parameter

Did you read the whole thread on the Kodi forum? That was what was initially proposed, but a solution was far from reached. It was rightly pointed out that depreciating the `<mpaa>` field would break lots of skins so it needs to be retained for backwards compatibility. The consensus seemed to be that three new fields would be introduced to introduce a better standard, but that the `<mpaa>` field would remain as it is now so it doesn't break everything.

> 
> > Where do you see that? Because the Wiki clearly states that MPAA is for country specific certifications: https://kodi.wiki/view/NFO_files/Movies
> 
> for addon cases: https://kodi.wiki/view/InfoLabels https://codedocs.xyz/AlwinEsch/kodi/group__python___info_tag_video.html#gafdddb4046567a6dc1ea91f8ab4d424ee currently this confuses everyone because the property is named as "mpaa" and the descriptions OFC is referred to MPAA only

These parts of the documentation obviously need updating, as that's not the real world usage of this field any more. When the feature was initially developed, it was probably accurate, as (presumably) the person who developed it only did it for their personal use case and MPAA data would have been the only information which populated that field...

However, naming it `<mpaa>` was incredibly short-sighted. People in other countries are obviously going to want that feature and would need a place for their data to live. As `<mpaa>` is the only field related to certification data, it's obviously going to be re-used... This is the far better solution than adding `<bbfc>`  for GB, `<FSK>` for DE, `<EIRIN>` for JP, `<CNC>` for FR etc, etc, etc.

> this is nice thing! but i am saying that managing this amount of images inside the skin is not really good then the solution proposed to use a resource addon can be a good idea, and this saves us from manage/add images to each skin

Kodi's default skin "Confluence" (back in v16 days) was what inspired me to create this mod. Team Kodi were obviously OK with packaging the .png files into the default skin once-upon-a-time, so I did it the same way: https://github.com/xbmc/skin.confluence/tree/master/media/flagging/ratings. 

It does make sense to have a resource addon provide the images though, if I'm honest, as it sounds like a better way of handling things.

jjd-uk commented:
> When the feature was initially developed, it was probably accurate, as (presumably) the person who developed it only did it for their personal use case and MPAA data would have been the only information which populated that field...

For more likely when it was added an extremely long time ago (no sure exactly when it was added but probably way over 10 yesrs ago) the metadata sites we scrape from only returned US ratings, and it was then simply kept named as it is for backwards compatibility

pkscout commented:
> Did you read the whole thread on the Kodi forum? That was what was initially proposed, but a solution was far from reached. It was rightly pointed out that depreciating the `<mpaa>` field would break lots of skins so it needs to be retained for backwards compatibility. The consensus seemed to be that three new fields would be introduced to introduce a better standard, but that the `<mpaa>` field would remain as it is now so it doesn't break everything.

Just one piece of clarification.  Yes, the intent is to introduce new ListItems for Kodi 20, but the current field wouldn't remain permanently. It would be depreciated for Kodi 20 (with a note to skinners to stop using it) and then removed for Kodi 21.  It doesn't make any sense to maintain a barely useful field that, due to the way various scrapers work, can have an almost infinite variation of data. So the intent is to give skinners time to change and then, as I said, remove it.  If we aren't going to depreciate and remove the current field, then I don't see the reason to add new ones.

boredazfcuk commented:
> Just one piece of clarification. Yes, the intent is to introduce new ListItems for Kodi 20, but the current field wouldn't remain permanently. It would be depreciated for Kodi 20 (with a note to skinners to stop using it) and then removed for Kodi 21. It doesn't make any sense to maintain a barely useful field that, due to the way various scrapers work, can have an almost infinite variation of data. So the intent is to give skinners time to change and then, as I said, remove it. If we aren't going to depreciate and remove the current field, then I don't see the reason to add new ones.

The reason for keeping it is backwards compatibility. There are a considerable amount of users who use the "Local Information Only" scraper as their preferred option as they manage their .nfo files using applications such as MediaElch/tinyMediaManager/EmberMediaManager/nfoMaker. This allows synchronisation of metadata across multiple clients in addition to Kodi, such as Jellyfin/Emby/Plex etc.

The proposed changes are definitely what is needed, as the current way of doing it is a mess, but there are far wider implications that was mentioned in the forum thread. A huge number of users will have to re-scrape and re-write their entire set of .nfo files. It's not a problem which can be dealt with by just notifying Kodi skinners. It'll need buy-in from all the application developers listed above, for it not to be a code breaker. Maybe it will happen, maybe the time frame is a little ambitious.

pkscout commented:
If we wait for buy in from everybody it will never happen. Assuming an annual Kodi release, two plus years should be plenty. No matter how long you wait, 20% of people won't change until forced too.

If the new fields get added for 20, then the scraper I maintain will use those and dump whatever it can in the old field. For 21 I'll plan on dropping the data in the old field. Or someone else can take over maintenance. If nothing changes, then the scrappers will keep putting inconsistent data in a misused field.

I kind of don't care anymore. I'm sorry I even brought it up.  It seems like many folks want things to change, but only as long as the change means everything stays the same.

wyrm65 commented:
pkscout,

as predicted this has turned into a huge pissing match. Take heart as this is exactly what happened to me 3 years ago. This time round we have the scrapper writers and skin writers on board all we need now is to have someone provide the fields so we can drop usage of MPAA fields.

For all I care we can call the fields Steve(country), Steve(ratings) and Steve(prefix), just call them something and get on with it.

As to the BBFC images in the resource file I created, I have used them under fair use provisions of copyright law. If the rights holders have an issue with their inclusion, it’s a five minute job to remove them. 

I will be reposting my code to access the resource file (which can  use ANY resource file that follows the naming scheme I have come up with) to forum later today. If people wish to continue using the old way of doing things, good luck to them. I for one have moved onto something better.

boredazfcuk commented:
> If we wait for buy in from everybody it will never happen. Assuming an annual Kodi release, two plus years should be plenty. No matter how long you wait, 20% of people won't change until forced too.

I dunno, I find users are pretty open to change. Identify and inform the end-users who will be affected. Bring on-board the people that will need to be assigned tasks and get them to do their prep work. The few that don't change are generally the ones that were oblivious to it.

> If the new fields get added for 20, then the scraper I maintain will use those and dump whatever it can in the old field. For 21 I'll plan on dropping the data in the old field. Or someone else can take over maintenance. If nothing changes, then the scrappers will keep putting inconsistent data in a misused field.

I honestly don't see the issue in leaving it there, but introducing the new ones too. There are already a bunch of depreciated, but still available, infoLabels. One more is hardly going to make a difference.
 
> I kind of don't care any more. I'm sorry I even brought it up. It seems like many folks want things to change, but only as long as the change means everything stays the same.

Don't get dejected about it, everyone wants this change, it's just that it's pretty much the equivalent of changing an industry standard, so it's going to take a while to find a solution that works from everyone's perspective.

>This time round we have the scrapper writers and skin writers on board all we need now is to have someone provide the fields so we can drop usage of MPAA fields.

This would make me think that this is the best time to push for it. Hopefully it doesn't fall by the wayside like it did 3 years ago.

> For all I care we can call the fields Steve(country), Steve(ratings) and Steve(prefix), just call them something and get on with it.

Or MPAA(country), MPAA(ratings) and MPAA(prefix)? haha, just kidding. I'm not particularly fussed, I think that Certification is already the widespread standard that's in use, so no issue with that. Anything with "Rating" should be avoided (parental/maturity), just because of the similarity with the "Ratings" system for the IMDb scores. Personally, I think that the term Classification is the most appropriate, as it's the term used on the MPAA and BBFC websites, and they're the ones that actually decide what classification the movie falls into anyway.

boredazfcuk commented:
Also, it would be good to get these guys on board too:

Media Elch: https://github.com/Komet/MediaElch/issues
tinyMediaManager: https://gitlab.com/users/sign_in
Ember Media Manager: https://forum.kodi.tv/showthread.php?tid=363079

I'm happy to open change requests for MediaElch/tinyMM  and chase them through. I'd need a full spec of what fields are required and data definitions. Ember's author is active on the Kodi Forum so I guess that getting onboard shouldn't be too difficult.

So far, I think that the proposed changes something along these lines:

Classification(Country): Mandatory - ISO 3166-1 Alpha3 Code
Classification(AgeRating): Mandatory - Rating of the work. e.g. U, PG, 15, R, NC-17 etc. TV ratings prefixed with "TV-"
Classification(Prefix): Optional - A prefix for the field e.g. "Rated ", "Rated: ", "UK: " etc.

A few of questions that jump out though... US Ratings already have TV- prefixed on them. Presumably we're not going to make that "TV-TV-MA"? But what if other countries prefix with "TV:" or suffix with "-TV"? We could end up with something like TV-MA-TV... So do we need to define what happens with special characters and such? Would it not be best to have a fourth field Classification(Medium) or Classification(Type) which can be set to "Movie" or "TV"?

pkscout commented:
The proposal is to use whatever rating string come back from the scraper. On TMDb at least, any country that has separate rating scales for TV shows and movies already adds TV. So we wouldn't add anything. If TV is in the rating that's fine. If it isn't, that's fine too.

KarellenX commented:
>TV ratings prefixed with "TV-"

I don't agree with that. Australia does not have official current classifications for TV that is prefixed "TV-". Our classification is the same as movies.

Like pkscout states, use whatever is returned from the scraper. Don't add anything to that as what the scraper returns is already correct.

https://www.classification.gov.au/

boredazfcuk commented:
> The proposal is to use whatever rating string come back from the scraper. On TMDb at least, any country that has separate rating scales for TV shows and movies already adds TV. So we wouldn't add anything. If TV is in the rating that's fine. If it isn't, that's fine too.

OK cool, I've edited my post.

> I don't agree with that. Australia does not have official current classifications for TV that is prefixed "TV-". Our classification is the same as movies.
> 
> Like pkscout states, use whatever is returned from the scraper. Don't add anything to that as what the scraper returns is already correct.
> 
> https://www.classification.gov.au/

Yeah, I think this was just my misunderstanding. I wasn't aware these prefixes were specified by the metadata source. I agree that using unmodified values is the best way forward. Less room for error.

wyrm65 commented:
Just bumping this as it again seems to have been forgotten.

Could one of the code monkeys PLEASE add at least a country code $INFO string so that we can move on with proper support of certification icons. Yes I know, pull requests are always welcome, but the last person you want messing with your code is me. The window for merges to the next Kodi release looks to be rapidly approaching closing and we still have not seen any resolution to the issue.
